R/summarystatsbayesianztest.R

```diff
@@ -0,0 +1,419 @@
+
+  return()
+}
+
+
+.bayesianZTestsDependencies   <- c(
+  "dataType", "dataEs", "dataSe", "dataLCi", "dataUCi",
+  "hypothesis", "priorMean", "priorSd"
+  )
+.bayesianZTestParseOptions       <- function(options) {
+
+  options[["dataEs"]]  <- jaspBase:::.parseRCodeInOptions(options[["dataEs"]])
+  options[["dataSe"]]  <- jaspBase:::.parseRCodeInOptions(options[["dataSe"]])
+  options[["dataLCi"]] <- jaspBase:::.parseRCodeInOptions(options[["dataLCi"]])
+  options[["dataUCi"]] <- jaspBase:::.parseRCodeInOptions(options[["dataUCi"]])
+
+  options[["priorMean"]] <- jaspBase:::.parseRCodeInOptions(options[["priorMean"]])
+  options[["priorSd"]]   <- jaspBase:::.parseRCodeInOptions(options[["priorSd"]])
+
+  return(options)
+}
+.bayesianZTestGetData            <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["data"]]))
+    return()
+
+  data <- createJaspState()
+  data$dependOn(.bayesianZTestsDependencies)
+  jaspResults[["data"]] <- data
+
+  if(options[["dataType"]] == "esAndSe"){
+    data[["object"]] <- list(
+      "es" = options[["dataEs"]],
+      "se" = options[["dataSe"]]
+    )
+  }else if(options[["dataType"]] == "esAndCi"){
+    data[["object"]] <- list(
+      "es" = options[["dataEs"]],
+      "se" = (options[["dataUCi"]] - options[["dataLCi"]]) / (2 * stats::qnorm(p = 0.975))
+    )
+  }else if(options[["dataType"]] == "esAndCiLog"){
+    data[["object"]] <- list(
+      "es" = log(options[["dataEs"]]),
+      "se" = (log(options[["dataUCi"]]) - log(options[["dataLCi"]])) / (2 * stats::qnorm(p = 0.975))
+    )
+  }
+
+  return()
+}
+.bayesianZTestSummaryTable       <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["summaryTable"]]))
+    return()
+
+  bfTitle <- .getBayesfactorTitleSummaryStats(options[["bayesFactorType"]], "twoSided")
+
+  summaryTable <- createJaspTable(title = gettext("Model Summary"))
+  summaryTable$dependOn(c(.bayesianZTestsDependencies, "bayesFactorType", "defaultBf", "defaultBfType", "defaultBfSd"))
+  summaryTable$position <- 1
+  summaryTable$addColumnInfo(name = "es",    title = gettext("Effect size"),        type = "number")
+  summaryTable$addColumnInfo(name = "se",    title = gettext("Standard error"),     type = "number")
+  summaryTable$addColumnInfo(name = "bf",    title = bfTitle,                       type = "number")
+  summaryTable$addColumnInfo(name = "maxBf", title = gettextf("Max %1$s", bfTitle), type = "number")
+
+  jaspResults[["summaryTable"]] <- summaryTable
+
+  data <- jaspResults[["data"]][["object"]]
+  bf10 <- .bayesianZTestsComputeBf10(
+    x   = data[["es"]],
+    se  = data[["se"]],
+    mu  = options[["priorMean"]],
+    sigma = options[["priorSd"]],
+    alternative = switch(
+      options[["hypothesis"]],
+      "notEqualToTestValue"  = "two.sided",
+      "greaterThanTestValue" = "greater",
+      "lessThanTestValue"    = "less"
+    ))
+  maxBf <- 1/exp(-(data[["es"]]/data[["se"]])^2/2)
+
+
+  bf10 <- switch(
```

Comment:
Spotted by @vandenman: You can (should) use `jaspBase::.recodeBFtype()` for flipping the Bayes factors here